python/graphstorm/gconstruct/construct_graph.py

```diff
@@ -409,35 +462,109 @@ def parse_edge_data(in_file, feat_ops, src_id_col, dst_id_col, edge_type,
             feat_data[key] = val
     src_ids = data[src_id_col]
     dst_ids = data[dst_id_col]
-    assert node_id_map is not None
-    src_type, _, dst_type = edge_type
-    if src_type in node_id_map:
-        src_ids = np.array([node_id_map[src_type][sid] for sid in src_ids])
-    else:
-        assert np.issubdtype(src_ids.dtype, np.integer), \
-                "The source node Ids have to be integer."
-    if dst_type in node_id_map:
-        dst_ids = np.array([node_id_map[dst_type][did] for did in dst_ids])
-    else:
-        assert np.issubdtype(dst_ids.dtype, np.integer), \
-                "The destination node Ids have to be integer."
+    src_ids, dst_ids = map_node_ids(src_ids, dst_ids, edge_type, node_id_map,
+                                    skip_nonexist_edges)
     return (src_ids, dst_ids, feat_data)
 
-def create_id_map(ids):
-    """ Create ID map
+class IdentityMap:
+    """ Identity map
+
+    This is an identity map. It doesn't do any mapping on IDs.
+
+    Parameters
+    ----------
+    size : int
+        The map size.
+    """
+    def __init__(self, size):
+        self._size = size
+
+    def __len__(self):
+        return self._size
+
+    def map_id(self, ids):
+        """ Map the input IDs to the new IDs.
+
+        This is identity map, so we don't need to do anything.
+
+        Parameters
+        ----------
+        ids : tensor
+            The input IDs
+
+        Returns
+        -------
+        tuple of tensors : the tensor of new IDs, the location of the IDs in the input ID tensor.
+        """
+        return ids, np.arange(len(ids))
+
+    def get_key_vals(self):
+        """ Get the key value pairs.
+        """
+        return None
```

Comment:
What is the purpose of having this function if it just returns `None`? If we want to have the same exact interface for "IdentityMap" and "IdMap" they should probably belong to a class hierarchy with a common interface.